### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by jeswr**
References and aligns with the `DatasetCore` api per https://github.com/rdfjs/N3.js/issues/286#issuecomment-1100438631

### CODE REVIEW

**Comment by jeswr** [comment]
Pinging @rubensworks since you did a lot of work on that spec.

File: README.md [hunk 1]
```diff
@@ -297,12 +297,25 @@ store.addQuad(
...
-for (const quad of store.readQuads(namedNode('http://ex.org/Mickey'), null, null))
+for (const quad of store.match(namedNode('http://ex.org/Mickey'), null, null))
   console.log(quad);
 ```
 
+### [`DatasetCore` Interface](https://rdf.js.org/dataset-spec/#datasetcore-interface)
+This store adheres to the `DatasetCore` interface which exposes the following properties
+
+Attributes:
+ - `size` A non-negative integer that specifies the number of quads in the set.
```
Comment (TallTed) [inline README.md:308]:
```suggestion
 - `size` — A non-negative integer that specifies the number of quads in the set.
```

File: README.md [hunk 2]
```diff
@@ -297,12 +297,25 @@ store.addQuad(
...
+
+Methods:
+ - `add` Adds the specified quad to the dataset. Existing quads, as defined in `Quad.equals`, will be ignored.
+ - `delete` Removes the specified quad from the dataset.
+ - `has` Determines whether a dataset includes a certain quad.
+ - `match` Returns a new dataset that is comprised of all quads in the current instance matching the given arguments.
+ - `[Symbol.iterator]` Implements the iterator protocol to allow iteration over all `quads` in the dataset as in the example above.
```
Comment (TallTed) [inline README.md:315]:
```suggestion
 - `add` — Adds the specified quad to the dataset. Existing quads, as defined in `Quad.equals`, will be ignored.
 - `delete` — Removes the specified quad from the dataset.
 - `has` — Determines whether a dataset includes a certain quad.
 - `match` — Returns a new dataset that is comprised of all quads in the current instance matching the given arguments.
 - `[Symbol.iterator]` — Implements the iterator protocol to allow iteration over all `quads` in the dataset as in the example above.
```

**Review by TallTed — CHANGES_REQUESTED** [review]
Backtick codefences are insufficient division for many screen readers and other accommodations for those with vision issues. Em-dash or similar dividers are recommended for all such "`term` — definition or description" line items.

**Comment by jeswr** [comment]
Pinging @RubenVerborgh for review

**Pull request merged**